File: Documentation/6.0/Samples/java/src/test/java/net/ravendb/ClientApi/Session/Querying/Debugging/IncludeQueryTimings.java
package net.ravendb.ClientApi.Session.Debugging;

import net.ravendb.client.documents.DocumentStore;
import net.ravendb.client.documents.IDocumentStore;
import net.ravendb.client.documents.queries.timings.QueryTimings;
import net.ravendb.client.documents.session.IDocumentQuery;
import net.ravendb.client.documents.session.IDocumentSession;
import net.ravendb.client.primitives.Reference;
import net.ravendb.northwind.Product;

import java.util.List;
import java.util.Map;

public class IncludeQueryTimings {

    public interface IFoo<T> {
        //region timing_1
        IDocumentQueryCustomization timings(Reference<QueryTimings> timings);
        //endregion
    }

    public void explain() {
        try (IDocumentStore store = new DocumentStore()) {
            try (IDocumentSession session = store.openSession()) {
                //region timing_2
                Reference<QueryTimings> timingsRef = new Reference<>();
                List<Product> resultsWithTimings = session.advanced().documentQuery(Product.class)
                    .timings(timingsRef)
                    .search("Name", "Syrup")
                    .toList();

                Map<String, QueryTimings> timingsMap = timingsRef.value.getTimings();
                //endregion
            }
        }
    }
}

File: Documentation/6.0/Samples/php/ClientApi/Session/Querying/Debugging/IncludeQueryTimings.php
<?php

use RavenDB\Documents\Queries\Timings\QueryTimings;
use RavenDB\Documents\Session\DocumentQueryInterface;
use RavenDB\Samples\Infrastructure\DocumentStoreHolder;
use RavenDB\Samples\Infrastructure\Orders\Product;

interface FooInterface
{
    # region syntax
    function timings(QueryTimings &$timings): DocumentQueryInterface;
    # endregion
}

class IncludeQueryTimings
{
    public function examples(): void
    {
        $store = DocumentStoreHolder::getStore();
        try {
            $session = $store->openSession();
            try {
                # region timing_2
                $timings = new QueryTimings();

                /** @var array<Product> $resultsWithTimings */
                $resultsWithTimings = $session->advanced()->documentQuery(Product::class)
                    ->timings($timings)
                    ->search("Name", "Syrup")
                    ->toList();

                /** @var array<QueryTimings> $timingsMap */
                $timingsMap = $timings->getTimings();
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
